src/test/datascience/notebook/controllerDefaultService.ts

```diff
@@ -60,8 +60,8 @@ export class ControllerDefaultService {
             if (controller || this.isWeb) {
                 return controller;
             }
-            traceVerbose('No default remote controller, hence returning the active interpreter');
-            return createActiveInterpreterController(viewType, resource, this.interpreters, this.registration);
+            // This should never happen.
+            throw new Error('No default remote controller, hence returning the active interpreter');
```

Comment:
Tests should fail instead of running successfully under the wrong conditions.
If we're running remote tests, then we MUST run against a remote kernel
Previously we'd just log a message and tests would end up running against a local kernel, and thats completely wrong.